var/spack/repos/builtin/packages/netcdf-fortran/package.py

```diff
@@ -132,6 +137,32 @@ def configure_args(self):
                 # not run by default and explicitly disabled above. To avoid the
                 # configuration failure, we set the following cache variable:
                 config_args.append("ac_cv_func_MPI_File_open=yes")
+            if "~shared" in self.spec:
```

Comment:
We will never be able to get lines 140 to 165 merged into the authoritative spack repository. Each of these packages should identify its own library dependencies and flags?